src/main/java/core/basesyntax/dao/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Set;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public boolean setBalance(Fruit fruit, int quantity) {
+        Storage.fruits.put(fruit, quantity);
+        return Storage.fruits.get(fruit) == quantity;
+    }
+
+    @Override
+    public boolean add(Fruit fruit, int quantity) {
+        int oldValue = Storage.fruits.get(fruit);
+        int newValue = oldValue + quantity;
+        Storage.fruits.put(fruit, newValue);
+        return Storage.fruits.get(fruit) == newValue;
+    }
+
+    @Override
+    public boolean subtract(Fruit fruit, int quantity) {
+        int oldValue = Storage.fruits.get(fruit);
+        int newValue = oldValue - quantity;
+        Storage.fruits.put(fruit, newValue);
+        return Storage.fruits.get(fruit) == newValue;
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage to ensure it is positive. Throw a RuntimeException if the balance is negative. This is to handle the Purchase operation correctly.